packages/protocol/contracts/authorizers/Api3ClientRrpAuthorizer.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./MetaAdminnable.sol";
+import "./ClientRrpAuthorizer.sol";
+import "./interfaces/IApi3ClientRrpAuthorizer.sol";
+
+/// @title Authorizer contract that whitelists clients where the API3 DAO is
+/// the metaAdmin
+contract Api3ClientRrpAuthorizer is
+    ClientRrpAuthorizer,
+    MetaAdminnable,
+    IApi3ClientRrpAuthorizer
+{
+    /// @notice Authorizer contracts use `authorizerType` to signal their type
+    uint256 public constant override AUTHORIZER_TYPE = 2;
+
+    /// @param metaAdmin_ Address that will be set as the API3 metaAdmin
+    constructor(address metaAdmin_) MetaAdminnable(metaAdmin_) {}
+
+    /// @notice Called to get the rank of an admin for an adminned entity
+    /// @dev Overriden to use metaAdminned ranks
+    /// @param adminnedId ID of the entity being adminned
+    /// @param admin Admin address whose rank will be returned
+    /// @return Admin rank
+    function getRank(bytes32 adminnedId, address admin)
+        public
+        view
+        override(MetaAdminnable, RankedAdminnable)
```

Comment:
So to me this seems to be more of a safety measure than anything. I found a medium article detailing some usecases for the override(Base1, Base2) inheritance, but basically if you're inheriting from multiple base contracts _that are unrelated_ then you must explicitly state which contracts you are overriding. In our case, the contracts are in a multiple inheritance structure so they know which functions to override, however it is better from a security perspective to directly assign them in the override call. So that there's no possibility of someone inheriting this class and being able to modify other contracts they should not be able to.

Reply:
Yeah I tried to be overly explicit